apps/system/js/base_module.js

```diff
@@ -111,6 +103,24 @@
   BaseModule.STATES = [];
 
   var SubmoduleMixin = {
+    // We could dynamically add and load new modules on demand.
+    loadWhenIdle: function(modules, time) {
+      return new Promise(function(resolve) {
+        this.constructor.SUB_MODULES =
+          this.constructor.SUB_MODULES.concat(modules);
+        var self = this;
+        var idleObserver = {
+          time: time || 10,
+          onidle: function() {
```

Comment:
Yes, it's intended. And even 10secs is still slow..I am thinking of a way to improve it if necessary.